File: modules/PosTagger.lua
require 'torch'
require 'nn'

local PosTagger = torch.class('nn.PosTagger')

-- Abstract base class for our PosTagger
function PosTagger:__init()
end

function PosTagger:train(taggedSentences)
  error('Not yet implemented!')
end

function PosTagger:validate(taggedSentences)
  error('Not yet implemented!')
end

function PosTagger:tag(sentence)
  error('Not yet implemented!')
end

function PosTagger:scoreTagging(taggedSentence)
  error('Not yet implemented!')
end

File: theano_port/PosTagger.py

class PosTagger(object):

  def train(self, tagged_sentences, learning_rate, iterations):
    raise NotImplementedError()

  def validate(self, tagged_sentences):
    raise NotImplementedError()

  def tag(self, sentence):
    raise NotImplementedError()

  def score_tagging(self, tagged_sentence):
    raise NotImplementedError()
